core/src/main/java/org/stellar/anchor/sep6/Sep6Service.java

```diff
@@ -1,38 +1,118 @@
+      Sep6Config sep6Config,
+      AssetService assetService,
+      Sep6TransactionStore txnStore,
+      EventService eventService) {
     this.sep6Config = sep6Config;
     this.assetService = assetService;
     this.txnStore = txnStore;
+    this.eventSession =
+        eventService.createSession(this.getClass().getName(), EventService.EventQueue.TRANSACTION);
     this.infoResponse = buildInfoResponse();
   }
 
   public InfoResponse getInfo() {
     return infoResponse;
   }
 
+  public GetDepositResponse deposit(Sep10Jwt token, GetDepositRequest request)
+      throws AnchorException {
+    // Pre-validation
+    if (token == null) {
+      throw new SepNotAuthorizedException("missing token");
+    }
+    if (request == null) {
+      throw new SepValidationException("missing request");
+    }
+    if (!request.getAccount().equals(token.getAccount())) {
+      throw new SepNotAuthorizedException("account does not match token");
+    }
+
+    AssetInfo asset = assetService.getAsset(request.getAssetCode());
+    if (asset == null || !asset.getDeposit().getEnabled() || !asset.getSep6Enabled()) {
+      throw new SepValidationException(
+          String.format("invalid operation for asset %s", request.getAssetCode()));
+    }
+
+    try {
+      KeyPair.fromAccountId(request.getAccount());
+    } catch (RuntimeException ex) {
+      throw new SepValidationException(String.format("invalid account %s", request.getAccount()));
+    }
+    Memo memo = MemoHelper.makeMemo(request.getMemo(), request.getMemoType());
+    String id = SepHelper.generateSepTransactionId();
+
+    Sep6TransactionBuilder builder =
+        new Sep6TransactionBuilder(txnStore)
+            .id(id)
+            .transactionId(id)
+            .status(SepTransactionStatus.INCOMPLETE.toString())
+            .kind(Sep6Transaction.Kind.DEPOSIT.toString())
+            .type(request.getType())
+            .assetCode(request.getAssetCode())
+            .assetIssuer(asset.getIssuer())
+            .amountOut(request.getAmount())
+            .amountOutAsset(asset.getAssetName())
+            .startedAt(Instant.now())
+            .sep10Account(token.getAccount())
+            .sep10AccountMemo(token.getAccountMemo())
+            .toAccount(request.getAccount());
+
+    if (memo != null) {
+      builder.memo(memo.toString());
+      builder.memoType(SepHelper.memoTypeString(MemoHelper.memoType(memo)));
+    }
+
+    Sep6Transaction txn = builder.build();
+    txnStore.save(txn);
+
+    eventSession.publish(
+        AnchorEvent.builder()
+            .id(UUID.randomUUID().toString())
+            .sep("6")
+            .type(AnchorEvent.Type.TRANSACTION_CREATED)
+            .transaction(TransactionHelper.toGetTransactionResponse(txn))
+            .build());
+
+    return GetDepositResponse.builder()
+        .how("Check the transaction for more information about how to deposit.")
+        .id(txn.getId())
+        .build();
```

Comment:
The `200` response in SEP-6 is meant to be a "no additional info necessary" response, but you're using the `how` attribute to guide the developer to checking the transaction for more information. We could go with this approach, but I think we would need to update the SEP to clarify that 200 doesn't mean "ready to receive funds".

I think we should consider using the [customer_info_needed](https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0006.md#3-customer-information-status) response instead. We would need to add the ID attribute to the response, but this response type is more accurate to the situation. What do you think?

Reply:
That's a good idea. Thanks for the suggestion!

Reply:
Hmm actually, I think I like your approach more 😅. I don't think it makes sense to return 403 and a transaction ID in the response body.